editor/debugger/script_editor_debugger.cpp

```diff
@@ -530,13 +530,24 @@ void ScriptEditorDebugger::_parse_message(const String &p_msg, const Array &p_da
 		error->set_text(1, error_title);
 		tooltip += " " + error_title + "\n";
 
+		// Find the language of the error's source file.
+		String source_language_name = "C++"; // Default value is the old hard-coded one.
+		const String source_file_extension = oe.source_file.get_extension();
+		for (int i = 0; i < ScriptServer::get_language_count(); ++i) {
+			ScriptLanguage *script_language = ScriptServer::get_language(i);
+			if (source_file_extension == script_language->get_extension()) {
+				source_language_name = script_language->get_name();
+				break;
+			}
+		}
+
 		if (!oe.error_descr.is_empty()) {
 			// Add item for C++ error condition.
 			TreeItem *cpp_cond = error_tree->create_item(error);
-			cpp_cond->set_text(0, "<" + TTR("C++ Error") + ">");
+			cpp_cond->set_text(0, "<" + source_language_name + " " + TTR("Error") + ">");
```

Comment:
This might break the "Open C++ Source on GitHub" right click option. See https://github.com/godotengine/godot/pull/66977.
You should also see if this option is shown for C# and what happens when it's selected.

Reply:
Still works fine for C++ sources (nothing basically didn't change for them). For C# we used to open a 404 and now shows the toast error. It's very weird to open the right click menu on C# files anyway because you usually have an external editor set, and even the right click opens the file in the external editor. Might be a good thing to fix in a subsequent PR.